### PR TITLE
Add some type hints

### PR DESCRIPTION
**Pull request by cydave**
Work in progress for #92

### CODE REVIEW

File: stumpy/floss.py [hunk 1]
```diff
@@ -139,6 +155,9 @@ def _cac(I, L, bidirectional=True, excl_factor=5, custom_iac=None, seed=0):
         A custom idealized arc curve (IAC) that will used for correcting the
         arc curve
 
+    seed : int
+        TODO
```
Comment (cydave) [inline stumpy/floss.py]: I wasn't sure on what might be a good description for this.
But I added it just for you :)

File: stumpy/floss.py [hunk 2]
```diff
@@ -73,6 +79,9 @@ def _iac(
     n_samples : int
         Number of distribution samples to draw during each iteration
 
+    seed : int
+        TODO
```
[the remark this answers is not in the document]
Comment (seanlaw) [reply stumpy/floss.py]: Great observation! I can add the description afterward

File: stumpy/floss.py (hunk 1 above)
Comment (seanlaw) [reply stumpy/floss.py]: Great observation! I can add the description afterward

**Comment by seanlaw** [comment]
@cydave Is this ready for review or should I hold off?

**Comment by cydave** [comment]
This should be ready to go.

File: stumpy/stump.py [hunk 3]
```diff
@@ -53,6 +65,8 @@ def _get_first_stump_profile(start, T_A, T_B, m, excl_zone, M_T, Σ_T, ignore_tr
     I : int64
         Matrix profile index for the window with index equal to `start`
     """
+    # TODO: return value of I is not simply an int64, looks like it's a tuple.
+    # it's either: (np.ndarray, int, int) or (np.ndarray, np.ndarray, np.ndarray)
```
Comment (seanlaw) [inline stumpy/stump.py]: You are right. The docstring is out of date. The return value of `I` should be a tuple of 3x `int64` (e.g., (3, 19, 20) or (10, -1, -1) ). The docstring should probably say something like:

```
I : Tuple[int64, int64, int64]
        Matrix profile index, left matrix profile index, and right matrix profile index for the window with index equal to `start`. The left and right matrix profile indices are automatically set to `-1` for self-joins (i.e., when `ignore_trivial` is set to `True`.
```

@cydave Would you mind updating the docstring (it's not obvious to me how the tuple should be represented there) and the type hint?

File: stumpy/gpu_stump.py [hunk 4]
```diff
@@ -154,23 +155,23 @@ def _compute_and_update_PI_kernel(
...
+    Σ_T: np.ndarray,
+    QT: np.ndarray,
+    QT_first: np.ndarray,
+    μ_Q: np.ndarray,
+    σ_Q: np.ndarray,
+    k: int,
+    ignore_trivial: bool = True,
+    range_start: int = 1,
+    threads_per_block: int = THREADS_PER_BLOCK,
+    device_id: int = 0,
```
Comment (seanlaw) [inline stumpy/gpu_stump.py]: I have some yet-to-be-committed updates where `device_id` can either be a single `int` (as it is shown here) or a list of `int`. Can you please tell me what this line should be if `device_id` can be either a single `int` or a list of `int`?  This way, I can update this on my own after your PR.

Comment (cydave) [reply stumpy/gpu_stump.py]: device_id: Union[int, List[int]] = 0
I think. Union => https://mypy.readthedocs.io/en/latest/kinds_of_types.html#union-types

File: stumpy/stump.py (hunk 3 above)
Comment (cydave) [reply stumpy/stump.py]: This looks to me like:
```
Tuple[np.int64, np.int64, np.int64] (if you require a numpy.int64, if you don't care too much about the specific type you could use Tuple[int, int, int]  :)

Try to find the actual type of the variables you're using and use those as the types you expect them to be in.

Comment (cydave) [reply stumpy/stump.py]: Although on second thought (after searching for a bit) the type might not need to be prefixed with `np.`. So the one you came up with appears to be the correct way to go about it.

Comment (seanlaw) [reply stumpy/stump.py]: It’s actually just a single element so I think int is perfectly fine

Comment (cydave) [reply stumpy/stump.py]: @seanlaw  I think the change at https://github.com/TDAmeritrade/stumpy/pull/93/commits/c37e1d7877e3e78fc6c47a88606b159c1e1fc606  should do it.

**Comment by seanlaw** [comment]
It looks like the entire `stumpy/chains.py` file was deleted?

**Comment by cydave** [comment]
@seanlaw  sorry about the mess of commits.

**Comment by seanlaw** [comment]
@cydave This may be too much to ask so feel free to push back and say, “no” but would you be open to just closing this PR and then taking the final version of your branch and creating a new PR with a single commit (with type hints and typos)?  As you pointed out, it would greatly help to reduce the number of commits to the main branch and keep the history more compact and searchable.

**Pull request closed without merging**